### ISSUE TITLE
Maintaining position (vertex references)

### ISSUE THREAD

**Issue by bmershon** [bmershon/d3-equidecompose#18]
In order to ensure intersections throughout the equidecomposition process, exact references are checked for vertices. **The following proposal and BUG was found in the process of implementing the halving procedure of the escalator method.**

`Polygon.translate` should probably operate in-place on the coordinates of the vertices so the Array object representing these tuples keep their reference. [Currently](https://github.com/bmershon/d3-equidecompose/blob/master/src/polygon.js#L14-L21), the reference is destroyed.

The following code seems to break things down the line, and I'm not sure why, but this in-place update seems like the right idea if we must check vertex references for EXACT equality (coordinates are floating-point, so coordinate comparison is fraught).

``` js
function translate(T) {
  for (let i = 0, n = this.length; i < n; i++) {
    let v = this[i].slice();
    this[i][0] = v[0] + T[0];
    this[i][1] = v[1] + T[1];
  }
  return this;
}
```

**Comment by bmershon** [comment]
Whoa boy. After scratching my head for a while after seeing **crazy #$@&** happen in the decomposition pipeline with the seemingly benign change outlined above, I realized that most of the **polygons generated "share" vertices with identical references**. If neighboring polygons experience different transforms... boom, you have a problem.

The real stinker is that it's nice to "know" what vertex you want to use in an intersection and simply track it through multiple transforms, knowing that wherever it lands, it still has the same reference and has been updated "in place". 

In the escalator method, the bottom right corner labeled **F** is one of those vertices that we need a reference to, because we cannot reliably perform an intersection by checking floating point coordinates.

One dumb idea is to search through the collection of polygons resulting from a cut to find **F** again before applying transforms.

![image](https://cloud.githubusercontent.com/assets/3190945/14764188/07d4ef84-097d-11e6-8ed5-de18466ba66b.png)
